website/addons/onedrive/utils.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+import time
+import logging
+from datetime import datetime
+
+from website.util import rubeus
+
+from website.addons.onedrive import settings
+
+
+logger = logging.getLogger(__name__)
+
+
+class OneDriveNodeLogger(object):
```

Comment:
Please see the generic NodeLogger defined in `website.addons.base.logger` and see if it will suit your purposes.

There's a corresponding property method in in the AddonOAuthNodeSettingsBase model that dynamically generates NodeLogger subclasses https://github.com/CenterForOpenScience/osf.io/blob/develop/website/addons/base/__init__.py#L803